Neos.Flow/Classes/Configuration/RouteConfigurationProcessor.php

```diff
@@ -193,16 +193,10 @@ protected function buildSubRouteConfigurations(array $routesConfiguration, array
                     $mergedSubRouteConfiguration['uriPattern'] = rtrim($this->replacePlaceholders($routeConfiguration['uriPattern'], [$subRouteKey => '']), '/');
                 }
                 if (isset($mergedSubRouteConfiguration['defaults'])) {
-                    foreach ($mergedSubRouteConfiguration['defaults'] as $key => $defaultValue) {
-                        $mergedSubRouteConfiguration['defaults'][$key] = $this->replacePlaceholders($defaultValue, $variables);
-                    }
+                    $mergedSubRouteConfiguration['defaults'] = $this->replacePlaceholders($mergedSubRouteConfiguration['defaults'], $variables);
                 }
                 if (isset($mergedSubRouteConfiguration['routeParts'])) {
-                    foreach ($mergedSubRouteConfiguration['routeParts'] as $routePartKey => $routePartValue) {
-                        foreach ($mergedSubRouteConfiguration['routeParts'][$routePartKey]['options'] ?? [] as $optionKey => $optionValue) {
-                            $mergedSubRouteConfiguration['routeParts'][$routePartKey]['options'][$optionKey] = $this->replacePlaceholders($optionValue, $variables);
-                        }
-                    }
+                    $mergedSubRouteConfiguration['routeParts'] = $this->replacePlaceholders($mergedSubRouteConfiguration['routeParts'], $variables);
```

Comment:
This will replace variables in all subkeys of 'routeParts', not only in the options - could this be an(other) issue? It could e.g. parameterize the handler name

Reply:
I thought about allowing this in the original change but didn't consider it valuable enough to make the code more complex.
But I don't think that it is a problem. I updated the issue description and added a comment (https://github.com/neos/flow-development-collection/issues/2142#issuecomment-704769881)